### DISCUSSION TITLE
limit() shouldn't accept i64, but u64, since negative LIMITs make no sense

### DISCUSSION THREAD

**Discussion by lecanard539** [diesel-rs/diesel#3824] [Q&A] [answered]
I think that limit() should not accept i64 but an unsigned number, such as u64, because negative limits (to the best of my knowledge) do not exist. In Postgres at least it will result in an error:

postgres=# select * from node limit -1;
ERROR:  LIMIT must not be negative

What does everyone else think? Should I open a bug report?

**Comment by weiznich** [comment, marked as answer]
That assumption is not correct. There are database systems where limit clauses with negative values have meaningful behavior. For example sqlite uses `LIMIT -1` to select all values, which can be used in combination with `OFFSET` clauses which in turn require the existent of a `LIMIT` clause there.
